src/qt/anybutton.cpp

```diff
@@ -24,18 +25,87 @@ class wxQtPushButton : public wxQtEventSignalHandler< QPushButton, wxAnyButton >
 {
 
 public:
+    virtual bool eventFilter(QObject *watched, QEvent *event);
     wxQtPushButton( wxWindow *parent, wxAnyButton *handler);
 
 private:
     void clicked(bool);
+    wxAnyButton *m_button;
 };
 
 wxQtPushButton::wxQtPushButton(wxWindow *parent, wxAnyButton *handler)
     : wxQtEventSignalHandler< QPushButton, wxAnyButton >( parent, handler )
 {
+    m_button = handler;
+    installEventFilter( this );
     connect(this, &QPushButton::clicked, this, &wxQtPushButton::clicked);
 }
 
+bool wxQtPushButton::eventFilter(QObject *watched, QEvent *event)
+{
+// TODO: Disabled control shouldn't get any events
+    QPushButton *button = qobject_cast<QPushButton *>( watched );
+    if( !button )
+        return false;
+    QIcon icon = button->icon();
+    if( event->type() == QEvent::HoverMove && button->isEnabled() )
+    {
+        if( m_button->GetStateBitmaps()[1].IsOk() )
+        {
+            button->setIcon( QIcon( *( m_button->GetStateBitmaps()[1].GetHandle() ) ) );
+//            return true;
+        }
+    }
+    else if( event->type() == QEvent::MouseButtonPress && button->isEnabled() )
+    {
+        if( m_button->GetStateBitmaps()[2].IsOk() )
+        {
+            button->setIcon( QIcon( *( m_button->GetStateBitmaps()[2].GetHandle() ) ) );
+        }
+    }
+    else if( event->type() == QEvent::FocusIn && button->isEnabled() )
+    {
+        if( m_button->GetStateBitmaps()[4].IsOk() )
+            button->setIcon( QIcon( *( m_button->GetStateBitmaps()[4].GetHandle() ) ) );
+    }
+    else if( event->type() == QEvent::EnabledChange )
+    {
+        if( !button->isEnabled() && m_button->GetStateBitmaps()[3].IsOk() )
+            button->setIcon( QIcon( *( m_button->GetStateBitmaps()[3].GetHandle() ) ) );
+        else
+        {
+            if( button->hasFocus() && m_button->GetStateBitmaps()[4].IsOk() )
+                button->setIcon( QIcon( *( m_button->GetStateBitmaps()[4].GetHandle() ) ) );
+            if( !button->hasFocus() && m_button->GetStateBitmaps()[0].IsOk() )
+                button->setIcon( QIcon( *( m_button->GetStateBitmaps()[0].GetHandle() ) ) );
```

Comment:
@catalinr 
It checks if the button has the focus

Reply:
Right, I missed the `!`. But why not continue with `else if` - `else` branching? It could make the code easier to read, especially when the key difference is a `!`.